File: apps/python/lib/chapter01/__main__.py
from lib.chapter01.kinoko_takenoko import (
    accuracy,
    load_people,
    predict_by_rule,
    split_features_and_labels,
)
from lib.dataset import data_dir


def main() -> None:
    people = load_people(data_dir() / "KvsT.csv")
    features, labels = split_features_and_labels(people)
    predictions = [predict_by_rule(f) for f in features]
    print(f"データ件数: {len(people)}")
    print(f"ルールによる判定の正解率: {accuracy(predictions, labels):.4f}")


if __name__ == "__main__":
    main()

File: apps/node/src/chapter01/main.ts
import { join } from "node:path";
import { dataDir } from "../dataset.ts";
import {
  accuracy,
  loadPeople,
  predictByRule,
  splitFeaturesAndLabels,
} from "./kinoko-takenoko.ts";

export function main(print: (line: string) => void = console.log): void {
  const people = loadPeople(join(dataDir(), "KvsT.csv"));
  const { features, labels } = splitFeaturesAndLabels(people);
  const predictions = features.map(predictByRule);
  print(`データ件数: ${people.length}`);
  print(
    `ルールによる判定の正解率: ${accuracy(predictions, labels).toFixed(4)}`,
  );
}

// node src/chapter01/main.ts で直接実行したときだけ main を呼ぶ
if (import.meta.filename === process.argv[1]) {
  main();
}
